homeassistant/components/zwave/climate.py

```diff
@@ -447,19 +498,19 @@ def set_fan_mode(self, fan_mode):
     def set_hvac_mode(self, hvac_mode):
         """Set new target hvac mode."""
         _LOGGER.debug("Set hvac_mode to %s", hvac_mode)
-        if not self.values.mode:
+        if not self.values.primary:
             return
         operation_mode = self._hvac_mapping.get(hvac_mode)
         _LOGGER.debug("Set operation_mode to %s", operation_mode)
-        self.values.mode.data = operation_mode
+        self.values.primary.data = operation_mode
```

Comment:
Please ignore that if it's a bad idea. We can prevent the setting of a mode that is already active with something like the following function:

```python
    def update_mode(self, operation_mode):
        """Update the operation mode only if it is not already set."""
        if self.values.primary.data != operation_mode:
            self.values.primary.data == operation_mode
```

I noticed that my zwave thermostats move when an active mode is activated again.